### PR TITLE
ECF-RP-523: Report schools

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
We want the start page to the 'report schools' lead provider journey.

### Changes proposed in this pull request
Add a static page, controller, route, tests. Mostly as foundations for another multi-page flow. 

### Guidance to review
Lots of stuff is copy-pasted, so double check that naming makes sense. I'm also open to suggestions on how you would define the routes in this flow. 

### Testing
Added simple request specs and cucumber tests.

### Review Checks
- [x] All pages have automated accessibility checks via cypress
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
Log in as `lead-provider@example.com`. Click 'Find and add schools'. You should see the start page. The button should do nothing.

### CODE REVIEW

**Comment by kicferk1** [comment]
[Report schools you've recruited for the next academic year (+ bug #6)](https://trello.com/c/PkNv5tio/523-report-schools-youve-recruited-for-the-next-academic-year-bug-6)

File: spec/factories/cohorts.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 FactoryBot.define do
   factory :cohort do
-    start_year { Faker::Date.unique.between(from: "2020-01-01", to: "2099-12-31").year }
+    start_year { 2020 + Faker::Number.unique.between(from: 1, to: 79) }
```
Comment (ethanmills) [inline spec/factories/cohorts.rb:5]: Doh!

File: app/views/lead_providers/report_schools/start.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <h1 class="govuk-heading-l">
+      You can only report schools you've recruited for the cohort that starts in the next academic year (2021)
```
Comment (amazimbe) [inline app/views/lead_providers/report_schools/start.html.erb:4]: This needs to come from cohort.current.display_name I believe

File: app/views/lead_providers/report_schools/start.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    </h1>
+
+    <p>
+      To report a school to the current academic year, contact DfE support.
```
Comment (amazimbe) [inline app/views/lead_providers/report_schools/start.html.erb:8]: Why can't we just have it on 1 line?

File: spec/requests/lead_providers/report_schools_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Report schools spec", type: :request do
+  let(:user) { create(:user, :lead_provider) }
+
+  describe "GET /lead_providers/report-schools/start" do
```
Comment (amazimbe) [inline spec/requests/lead_providers/report_schools_spec.rb:8]: Can we update lead_providers to lead-providers

File: app/views/lead_providers/report_schools/start.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class="govuk-grid-row">
```
Comment (ethanmills) [inline app/views/lead_providers/report_schools/start.html.erb:1]: I think we're getting copy for this in the next half hour or so

File: spec/cypress/integration/lead_providers/ReportSchools.feature [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Feature: Report Schools flow
+  Scenario: Visiting the start page
+    Given I am on "lead providers report schools start" page
+    Then the page should be accessible
```
Comment (callumacrae) [inline spec/cypress/integration/lead_providers/ReportSchools.feature:4]: Send percy a snapshot? 😄

Comment (ethanmills) [inline spec/cypress/integration/lead_providers/ReportSchools.feature:4]: I think I'd prefer this to start on the dashboard, to prove the journey joins up. Is everything in place to do that?

Comment (kicferk1) [reply spec/cypress/integration/lead_providers/ReportSchools.feature]: Well, not quite, but it wasn't too hard to set it up.

File: app/views/lead_providers/report_schools/start.html.erb (hunk 5 above)
Comment (kicferk1) [reply app/views/lead_providers/report_schools/start.html.erb:1]: I've updated the copy on the page I added.

**Review by amazimbe — APPROVED** [review]

File: app/views/dashboard/show.html.erb [hunk 7]
```diff
@@ -7,14 +7,13 @@
...
-<!--            TODO Fix after Ethan code merge for lead providers-->
-            <%= govuk_link_to "Find and add schools", nil %>
+            <%= govuk_link_to "Find and add schools", start_lead_providers_report_schools_path %>
           </h2>
           <p class="govuk-body">Search for schools you want to recruit and add schools you have partnered with.</p>
         </div>
         <div class="index-group">
           <h2 class="govuk-heading-m">
-            <%= govuk_link_to "Check your schools for #{@lead_provider.cohorts.find_by(start_year: Time.zone.now.year).start_year}", lead_providers_your_schools_path(selected_cohort_id: @lead_provider.cohorts.find_by(start_year: Time.zone.now.year).id) %>
+            <%= govuk_link_to "Check your schools for #{@lead_provider.cohorts.find_by(start_year: Time.zone.now.year)&.start_year}", lead_providers_your_schools_path(selected_cohort_id: @lead_provider.cohorts.find_by(start_year: Time.zone.now.year)&.id) %>
```
Comment (BroiSatse) [inline app/views/dashboard/show.html.erb:16]: Would `@lead_provider.cohorts.current.&id` work here as well?

File: spec/factories/cohorts.rb [hunk 8]
```diff
@@ -2,6 +2,6 @@
 
 FactoryBot.define do
   factory :cohort do
-    start_year { Faker::Date.unique.between(from: "2020-01-01", to: "2099-12-31").year }
+    start_year { Faker::Number.unique.between(from: 2021, to: 2100) }
```
Comment (BroiSatse) [inline spec/factories/cohorts.rb:5]: :champagne:

File: app/views/dashboard/show.html.erb (hunk 7 above)
Comment (kicferk1) [reply app/views/dashboard/show.html.erb:16]: Ehh, the cohort would be missing if we messed up seeding. I will look into making it nicer.

File: app/views/lead_providers/report_schools/start.html.erb [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  <div class="govuk-grid-column-two-thirds">
+    <h1 class="govuk-heading-xl">You can only confirm schools for the cohort that starts in the next academic year
+      (<%= Cohort.current.display_name %>).</h1>
+
+    <p>To report a school to the current academic year, contact DfE support.</p>
```
Comment (ethanmills) [inline app/views/lead_providers/report_schools/start.html.erb:6]: I think this line needs to go, right?

File: app/views/lead_providers/report_schools/start.html.erb [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+
+    <%= govuk_link_to "Continue", '#', class: 'govuk-button' %>
```
Comment (ethanmills) [inline app/views/lead_providers/report_schools/start.html.erb:8]: This can be `button: true`

**Review by ethanmills — APPROVED** [review]

**Pull request merged**